connector/wrappers/medchain/client.go

```diff
@@ -0,0 +1,143 @@
+package medchain
+
+import (
+	"sync/atomic"
+
+	"github.com/ldsec/medchain/contracts"
+	"go.dedis.ch/cothority/v3/byzcoin"
+	"go.dedis.ch/cothority/v3/byzcoin/bcadmin/lib"
+	"go.dedis.ch/cothority/v3/darc"
+	"go.dedis.ch/protobuf"
+	"golang.org/x/xerrors"
+)
+
+// QueryStatus denotes the status of a query.
+type QueryStatus string
+
+const (
+	// QuerySuccessStatus denotes a "success" status
+	QuerySuccessStatus QueryStatus = contracts.QuerySuccessStatus
+	// QueryFailedStatus denotes a "failed" status
+	QueryFailedStatus QueryStatus = contracts.QueryFailedStatus
+)
+
+// Client is used to interact with medchain.
+type Client struct {
+	bcClient *byzcoin.Client
+	signer   darc.Signer
+	counter  uint64
+}
+
+// NewClient creates a new Client by parsing a Byzcoin configuration file
+// at `path` and uses `signer` for signing the transactions
+func NewClient(path string, signer darc.Signer) (*Client, error) {
+	_, bc, err := lib.LoadConfig(path)
+	if err != nil {
+		return nil, xerrors.Errorf("failed to initialize byzcoin client: %v", err)
+	}
+
+	return &Client{bcClient: bc, signer: signer}, nil
+}
+
+// GetAuthorization returns the Authorization a user has for a given query in a project.
+func (c *Client) GetAuthorization(projectInstID byzcoin.InstanceID, userID, queryID, queryDef string) (*contracts.Authorization, byzcoin.InstanceID, error) {
+	instr := byzcoin.Instruction{
+		InstanceID: projectInstID,
+		Spawn: &byzcoin.Spawn{
+			ContractID: contracts.QueryContractID,
+			Args: []byzcoin.Argument{{
+				Name:  contracts.QueryUserIDKey,
+				Value: []byte(userID),
+			}, {
+				Name:  contracts.QueryQueryIDKey,
+				Value: []byte(queryID),
+			}, {
+				Name:  contracts.QueryQueryDefinitionKey,
+				Value: []byte(queryDef),
+			}},
+		},
+		SignerCounter: []uint64{atomic.AddUint64(&c.counter, 1)},
```

Comment:
we might want to use `c.bcClient.GetSignerCounters()` to get the signer's counter